src/lib.rs

```diff
@@ -100,6 +100,7 @@
 //! 
 //!   - `slab` providing the `Slab` collection.
 //!   - `smallvec` providing the `SmallVec` collection.
+#![feature(generic_associated_types)]
```

Comment:
Since I would prefer for this crate to remain usable with stable rust, we'll need to wait for GATs stabilization and remove this before merging.